File: Data Structures & Algorithms/permutations/submission-0.java
class Solution {
    List<List<Integer>> l = new ArrayList<>();
    private void helper(int[] nums, int idx){
        if(idx == nums.length){
            l.add(Arrays.stream(nums).boxed().collect(Collectors.toList()));
            return;
        }

        for(int i = idx ; i < nums.length ; i++){
            int k = nums[i];
            nums[i] = nums[idx];
            nums[idx] = k;
            helper(nums, idx+1);
            k = nums[i];
            nums[i] = nums[idx];
            nums[idx] = k;
        }
    }
    public List<List<Integer>> permute(int[] nums) {
        l = new ArrayList<>();
        helper(nums, 0);
        return l;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int i = nums[0];
        nums[0] = -1;
        while(nums[i] != -1){
            int t = nums[i];
            nums[i] = -1;
            i = t;
        }
        return i;
    }
};
